unlock-app/src/components/interface/locks/Settings/forms/UpdateReferralFee.tsx

```diff
@@ -16,82 +18,115 @@ interface UpdateReferralFeeProps {
   const { getWalletService } = useAuth()
 
   const {
     register,
     handleSubmit,
-    setValue,
+    reset,
     formState: { errors },
-  } = useForm<FormProps>()
+  } = useForm<FormProps>({
+    defaultValues: { referralFeePercentage: 0, referralAddress: '' },
+  })
 
   const setReferrerFee = async (fields: FormProps) => {
     const walletService = await getWalletService(network)
     await walletService.setReferrerFee({
       lockAddress,
-      address: ZERO,
+      address: fields?.referralAddress,
       feeBasisPoint: fields?.referralFeePercentage * 100,
     })
   }
 
-  const getReferrerFees = async () => {
-    return await web3Service.referrerFees({
-      lockAddress,
-      network,
-      address: ZERO,
-    })
+  const getLock = async () => {
+    const service = new SubgraphService()
+    return service.lock(
+      {
+        where: {
+          id: lockAddress,
+        },
+      },
+      {
+        network,
+      }
+    )
   }
 
   const setReferrerFeeMutation = useMutation(setReferrerFee)
 
   const onSubmit = async (fields: FormProps) => {
-    const setReferrerFeePromise = setReferrerFeeMutation.mutateAsync(fields)
+    setIsResolvingAddress(true)
+
+    const lowerCasedAddress = fields.referralAddress.toLowerCase()
+
+    // This will return resolved ens name or existing address
+    const resolvedAddress = await getAddressForName(lowerCasedAddress)
+
+    // We are checking if the resolved address is valid
+    const isValidAddress = ethers.utils.isAddress(resolvedAddress.toLowerCase())
+
+    // If not a valid address return early and give an error toaster
+    if (!isValidAddress) {
+      ToastHelper.error(`Referrer address is not valid, please check the value`)
+      setIsResolvingAddress(false)
+      return
+    }
+
+    setIsResolvingAddress(false)
+
+    const setReferrerFeePromise = setReferrerFeeMutation.mutateAsync({
+      ...fields,
+      referralAddress: ethers.utils.getAddress(resolvedAddress.toLowerCase()),
+    })
 
     await ToastHelper.promise(setReferrerFeePromise, {
-      loading: 'Updating referrer fee',
+      loading: 'Setting new referrer',
       error: 'Failed to update the values, please try again.',
-      success: 'Referrer fee updated.',
+      success: 'New referrer set',
     })
+
+    reset()
   }
 
-  const { isLoading, data: referralFeePercentage } = useQuery(
-    ['getReferrerFees', lockAddress, network, setReferrerFeeMutation.isSuccess],
-    async () => getReferrerFees()
+  const { isLoading, data: referralFeesData } = useQuery(
+    ['getLock', lockAddress, network, setReferrerFeeMutation.isSuccess],
+    async () => getLock()
   )
 
-  useEffect(() => {
-    setValue('referralFeePercentage', (referralFeePercentage ?? 0) / 100)
-    setReferralFee((referralFeePercentage ?? 0) > 0)
-  }, [referralFeePercentage])
+  // @ts-ignore
+  const referralFees = (referralFeesData?.referrer as ReferrerFee[]) || []
```

Comment:
It's not good to use `@ts-ignore` but i do this for now since when i try to build my `subgraph` fields it has this error:
 
![referrer_thegraph](https://github.com/unlock-protocol/unlock/assets/136458308/185002cb-be57-4d50-8bca-a7d1dc11f1bd)

The easiest way @julien51 is if you can update the subgraph on all networks so i can rebuild it on my end